File: solutions/2882-ways-to-express-an-integer-as-sum-of-powers/ways-to-express-an-integer-as-sum-of-powers.go
func numberOfWays(n int, x int) int {
	const mod = 1_000_000_007
	dp := make([]int, n+1)
	dp[0] = 1
	for i := 1; i <= n; i++ {
		val := int(math.Pow(float64(i), float64(x)))
		if val > n {
			break
		}
		for j := n; j >= val; j-- {
			dp[j] = (dp[j] + dp[j-val]) % mod
		}
	}
	return dp[n]
}

File: solutions/2882-ways-to-express-an-integer-as-sum-of-powers/ways-to-express-an-integer-as-sum-of-powers.cs
public class Solution {
    public int NumberOfWays(int n, int x) {
        const int MOD = 1_000_000_007;
        long[] dp = new long[n + 1];
        dp[0] = 1;
        for (int i = 1; i <= n; i++) {
            int val = (int)Math.Pow(i, x);
            if (val > n) {
                break;
            }
            for (int j = n; j >= val; j--) {
                dp[j] = (dp[j] + dp[j - val]) % MOD;
            }
        }
        return (int)dp[n];
    }
}